src/Symfony/Component/Form/FormConfigBuilder.php

```diff
@@ -642,12 +642,12 @@ final public static function validateName(?string $name): void
      * A name is accepted if it
      *
      *   * is empty
-     *   * starts with a letter, digit or underscore
      *   * contains only letters, digits, numbers, underscores ("_"),
      *     hyphens ("-") and colons (":")
+     *   * is not equal to "isindex"
      */
     final public static function isValidName(?string $name): bool
     {
-        return '' === $name || null === $name || preg_match('/^[a-zA-Z0-9_][a-zA-Z0-9_\-:]*$/D', $name);
+        return ('' === $name || null === $name || preg_match('/^[a-zA-Z0-9_\-:]*$/D', $name)) && 'isindex' !== $name;
```

Comment:
Why do we fordbid `isindex` now? This is a BC break if I don’t miss anything.

Reply:
Because `isindex` is not allowed in HTML5.

Yeah, it's a BC break. Any idea how to handle this case?

Reply:
Let‘s deprecate it in 7.1 and don’t accept it in 8.0.

Reply:
I think this is wrong though. Most of the Symfony form names could use `isindex` without issue. The HTML restriction would only forbid us to generate a _full name_ equal to `isindex`, while `root[isindex]` is perfectly valid in HTML (remember that the Symfony form name is not used directly as the HTML name, but is involved in generating it)

Reply:
Just adding that it's not only `isindex`, but also `_charset_`, see https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#attr-fe-name

> An [ASCII case-insensitive](https://infra.spec.whatwg.org/#ascii-case-insensitive) match for the name `_charset_` is special: if used as the name of a [Hidden](https://html.spec.whatwg.org/multipage/input.html#hidden-state-(type=hidden)) control with no [value](https://html.spec.whatwg.org/multipage/input.html#attr-input-value) attribute, then during submission the [value](https://html.spec.whatwg.org/multipage/input.html#attr-input-value) attribute is automatically given a value consisting of the submission character encoding.